package-res/resources/web/vizapi/VizController.js

```diff
@@ -716,29 +716,31 @@ pentaho.VizController.getRgbGradientFromMultiColorHex = function(value, min, max
   var steps = colors.length-1;
   var range = max-min;
 
+  var startStep;
+  var endStep;
   if(range <= 0) {
-    var start = colors.length-1;
-    var end = start;
+    startStep = colors.length-1;
+    endStep = startStep;
   } else {
-    var start = Math.floor(((value-min)/range) * steps);
-    var end = Math.ceil(((value-min)/range) * steps);
+    var inRange = ((value-min)/range);
+    startStep = Math.floor(inRange * steps);
+    endStep = Math.ceil(inRange * steps);
   }
-  var color1 = pentaho.VizController.convertToRGB(colors[start]);
-  var color2 = pentaho.VizController.convertToRGB(colors[end]);
+  var color1 = pentaho.VizController.convertToRGB(colors[startStep]);
+  var color2 = pentaho.VizController.convertToRGB(colors[endStep]);
 
-  var rangeMin = (start == 0) ? 1 : (start / steps) * max;
-  var rangeMax = (end / steps) * max;
-
-  var inRange
-  if(rangeMin == rangeMax){
-    inRange = 1;
+  var inStepRange;
+  if(startStep == endStep){
+    inStepRange = 0;
   } else {
-    inRange = (value-rangeMin)/(rangeMax-rangeMin);
+    var stepRangeMin = ((startStep / steps) * range) + min;
+    var stepRangeMax = ((endStep / steps) * range) + min;
+    inStepRange = (value-stepRangeMin)/(stepRangeMax-stepRangeMin);
```

Comment:
Personally, I find the names `inRange` and `inStepRange` is a bit obscure.
What about renaming the variable `inStepRange` to `scalingFactor` or something more suggestive?